lib/mayaUsd/render/MaterialXGenOgsXml/OgsFragment.cpp

```diff
@@ -423,6 +424,156 @@ std::string getOriginalName(const std::string& parameterName, const mx::NodeDefP
     return *it;
 }
 
+template <typename T>
+static bool
+parameterDiffersFrom(const mx::NodePtr& node, const std::string& paramName, T const& paramValue)
```

Comment:
Copy of code introduced in USD in https://github.com/PixarAnimationStudios/OpenUSD/pull/3073 but adapted to MaterialX.

Please node associated MaterialX issue: https://github.com/AcademySoftwareFoundation/MaterialX/issues/1946